api/ocm/extensions/accessmethods/git/method.go

```diff
@@ -0,0 +1,188 @@
+package git
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/mandelsoft/goutils/errors"
+	giturls "github.com/whilp/git-urls"
+
+	"ocm.software/ocm/api/credentials"
+	"ocm.software/ocm/api/credentials/builtin/git/identity"
+	"ocm.software/ocm/api/ocm/cpi/accspeccpi"
+	"ocm.software/ocm/api/ocm/internal"
+	techgit "ocm.software/ocm/api/tech/git"
+	"ocm.software/ocm/api/utils/accessio"
+	"ocm.software/ocm/api/utils/accessio/downloader/git"
+	"ocm.software/ocm/api/utils/accessobj"
+	"ocm.software/ocm/api/utils/blobaccess/blobaccess"
+	"ocm.software/ocm/api/utils/mime"
+	"ocm.software/ocm/api/utils/runtime"
+)
+
+const (
+	Type   = "git"
+	TypeV1 = Type + runtime.VersionSeparator + "v1"
+)
+
+func init() {
+	accspeccpi.RegisterAccessType(accspeccpi.NewAccessSpecType[*AccessSpec](Type, accspeccpi.WithDescription(usage)))
+	accspeccpi.RegisterAccessType(accspeccpi.NewAccessSpecType[*AccessSpec](TypeV1, accspeccpi.WithFormatSpec(formatV1), accspeccpi.WithConfigHandler(ConfigHandler())))
+}
+
+// AccessSpec describes the access for a GitHub registry.
+type AccessSpec struct {
+	runtime.ObjectVersionedType `json:",inline"`
+
+	// RepoURL is the repository URL
+	RepoURL string `json:"repoUrl"`
+
+	// Ref defines the hash of the commit
+	Ref string `json:"ref"`
+
+	// PathSpec is a path in the repository to download, can be a file or a regex matching multiple files
+	PathSpec string `json:"pathSpec"`
+}
+
+// AccessSpecOptions defines a set of options which can be applied to the access spec.
+type AccessSpecOptions func(s *AccessSpec)
+
+// New creates a new git registry access spec version v1.
+func New(url, ref string, pathSpec string, opts ...AccessSpecOptions) *AccessSpec {
+	s := &AccessSpec{
+		ObjectVersionedType: runtime.NewVersionedTypedObject(Type),
+		RepoURL:             url,
+		Ref:                 ref,
+		PathSpec:            pathSpec,
+	}
+	for _, o := range opts {
+		o(s)
+	}
+	return s
+}
+
+func (a *AccessSpec) Describe(internal.Context) string {
+	return fmt.Sprintf("git commit %s[%s]", a.RepoURL, a.Ref)
+}
+
+func (*AccessSpec) IsLocal(internal.Context) bool {
+	return false
+}
+
+func (a *AccessSpec) GlobalAccessSpec(accspeccpi.Context) accspeccpi.AccessSpec {
+	return a
+}
+
+func (*AccessSpec) GetType() string {
+	return Type
+}
+
+func (a *AccessSpec) AccessMethod(c internal.ComponentVersionAccess) (internal.AccessMethod, error) {
+	return accspeccpi.AccessMethodForImplementation(newMethod(c, a))
+}
+
+func newMethod(componentVersionAccess internal.ComponentVersionAccess, accessSpec *AccessSpec) (accspeccpi.AccessMethodImpl, error) {
+	u, err := giturls.Parse(accessSpec.RepoURL)
+	if err != nil {
```

Comment:
Great, that you implemented a general git access, also,

But unfortunately this should include more extensions to be implemented consistently:
- blob accesses
- access method
- input types (for the CLI based composition environment)
- the convenience functions in the `elements` package. (I personally would prefer to get rid of them) 

For the other accesses, we have implemented the blob access and then based the others on this common implementation.

Reply:
I implemented the blob access and the access method based on that blob access now. For the input types and the elements, thats still missing, still need a pointer